### PR TITLE
Adding XFS as a root filesystem type

### PR DESCRIPTION
**Pull request by AZaugg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Adding XFS support to the installer, as well as allow Mariner to boot XFS filesystems.

###### Change Log  

- Added XFS option to Mariner Installer
- Added fsck.xfs to initrd
- Added XFS module to grub

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Build a new VHD and booted, confirmed root filesystem was running XFS

### CODE REVIEW

**Comment by AZaugg** [comment]
Not sure, if we want to add a default mount option of bigtime=1 for any XFS mount.

**Comment by dmcilvaney** [comment]
Looks good, just two things:
1) Looks like Vince sniped you with an update to the initramfs package, so needs a rebase
2) Could you add xfsprogs to the [prerequisites doc](toolkit/docs/building/prerequisites.md)

File: toolkit/resources/manifests/package/toolchain_aarch64.txt [hunk 1]
```diff
@@ -575,6 +575,7 @@ util-linux-devel-2.37.4-5.cm2.aarch64.rpm
 util-linux-lang-2.37.4-5.cm2.aarch64.rpm
 which-2.21-8.cm2.aarch64.rpm
 which-debuginfo-2.21-8.cm2.aarch64.rpm
+xfsprogs-5.0.0-3.cm2.aarch64.rpm
```
Comment (dmcilvaney) [inline toolkit/resources/manifests/package/toolchain_aarch64.txt:578]: Drop this

File: toolkit/resources/manifests/package/toolchain_x86_64.txt [hunk 2]
```diff
@@ -267,12 +267,12 @@ npth-1.6-4.cm2.x86_64.rpm
 npth-debuginfo-1.6-4.cm2.x86_64.rpm
 npth-devel-1.6-4.cm2.x86_64.rpm
 ntsysv-1.20-3.cm2.x86_64.rpm
-openssl-1.1.1k-21.cm2.x86_64.rpm
-openssl-debuginfo-1.1.1k-21.cm2.x86_64.rpm
-openssl-devel-1.1.1k-21.cm2.x86_64.rpm
-openssl-libs-1.1.1k-21.cm2.x86_64.rpm
-openssl-perl-1.1.1k-21.cm2.x86_64.rpm
-openssl-static-1.1.1k-21.cm2.x86_64.rpm
```
Comment (dmcilvaney) [inline toolkit/resources/manifests/package/toolchain_x86_64.txt:275]: Looks like this needs a cleanup

**Review by dmcilvaney — CHANGES_REQUESTED** [review]

File: toolkit/resources/manifests/package/toolchain_aarch64.txt (hunk 1 above)
Comment (AZaugg) [reply toolkit/resources/manifests/package/toolchain_aarch64.txt]: Done

**Comment by dmcilvaney** [comment]
> Added @chalamalasetty who tracks the size of our ISOs as a reviewer.
> 
> @chalamalasetty - Looking at the dependencies for xfsprogs I think we already have them all in the image, so the only effect is xfsprogs itself. (4.27MB)

Validated with a low memory device, seems fine still.

**Pull request merged**